src/test/java/com/clickhouse/kafka/connect/sink/ClickHouseSinkTaskWithSchemaTest.java

```diff
@@ -213,7 +213,11 @@ public void detectUnsupportedDataConversions() {
 
         ClickHouseSinkTask chst = new ClickHouseSinkTask();
         chst.start(props);
-        assertThrows(DataException.class, () -> chst.put(sr), "Did not detect wrong date conversion ");
+        try {
+            chst.put(sr);
+        } catch (RuntimeException e) {
+            assertTrue(Utils.getRootCause(e) instanceof DataException, "Did not detect wrong date conversion ");
```

Comment:
Are these changes related? If not, let's not mix several changes in a single PR - it would greatly simplify the review.